### ISSUE TITLE
lsp-mode overrides custom flycheck checker

### ISSUE THREAD

**Issue by zjunze** [hlissner/doom-emacs#3730]
Hi there!

In my config, I explicitly set flycheck to use python-flake8 as the checker in python mode(see below), it used to work but it stopped working after I upgraded doom yesterday. Now lsp-mode is overriding my config and always set the checker as lsp. Not sure if it's a bug or my config is not ideal? Please provide some guidance. Thanks! :) 

```
(after! python
  (setq
        flycheck-checker 'python-flake8
        python-indent-offset 4
        flycheck-python-pylint-executable "pylint"
        flycheck-python-flake8-executable "flake8"
  )
)
```

**Comment by hlissner** [comment]
This appears to be a common misunderstanding about `after!` and buffer-local variables.

TL;DR, use this:

```elisp
(after! python
  (setq python-indent-offset 4
        flycheck-python-pylint-executable "pylint"
        flycheck-python-flake8-executable "flake8"))

(setq-hook! 'python-mode-hook flycheck-checker 'python-flake8)
```

----

+ `flycheck-checker` is a _buffer-local_ variable. Meaning its value can be different from buffer to buffer.
+ `(after! X BODY...)` evalutes `BODY` _once_, when `X` loads. It does not evaluate `BODY` in the context of any one buffer. If it happens to affect the python buffer you're working on at that moment, that is the _only_ buffer that will be affected by this change.
+ `setq-hook!` adds a hook to `MODE-hook` that will run _each time_ `MODE` is activated.

----

Please direct usage and configuration [to our Discord](https://discord.gg/qvGgnVx) in the future.

**Comment by zjunze** [comment]
Thanks for the info! I've tried the provided config but the checker is still explicitly set to lsp by the lsp-mode:

![Capture](https://user-images.githubusercontent.com/10176831/89931832-9d277000-dbc1-11ea-9884-89cbc13a2dcf.JPG)

I will use discord in the future. Thanks!
